neverbleed.c

```diff
@@ -505,10 +527,40 @@ void neverbleed_transaction_write(neverbleed_t *nb, neverbleed_iobuf_t *buf)
     iobuf_transaction_write(buf, thdata);
 }
 
+static void do_exdata_free_callback(void *parent, void *ptr, CRYPTO_EX_DATA *ad, int idx, long argl, void *argp)
+{
+    assert(ptr != NULL);
```

Comment:
```suggestion
    if (ptr == NULL)
        return;
```
Sorry I think it was actually correct to do a return here.

Applications may call RSA_new_method or EC_KEY_new_method on a different engine then invoke RSA_free or EC_KEY_free. If that happens, the free callback gets invoked by `ptr` will be NULL.

I think that's why we are seeing crashes in https://github.com/h2o/h2o/actions/runs/6306221925/job/17120995094.